modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/impl/KBArticleLocalServiceImpl.java

```diff
@@ -513,12 +513,15 @@ public KBArticle expireKBArticle(
 		KBArticle kbArticle = getLatestKBArticle(
 			resourcePrimKey, WorkflowConstants.STATUS_ANY);
 
-		kbArticle.setExpirationDate(new Date());
-
-		kbArticleLocalService.updateKBArticle(kbArticle);
-
 		serviceContext.setCommand(Constants.EXPIRE);
 
+		_updateKBArticle(
+			kbArticle.getUserId(), resourcePrimKey, kbArticle.getTitle(),
+			kbArticle.getContent(), kbArticle.getDescription(),
+			KBSectionEscapeUtil.unescapeSections(kbArticle.getSections()),
+			kbArticle.getSourceURL(), new Date(), kbArticle.getReviewDate(),
+			null, null, serviceContext);
+
```

Comment:
There is a little problem with this change. Now an article subscriber receives two notifications instead of one: an update notification and an expiration notification. Before only the expiration notification was sent.

I've seen that the update notification is raised by this call in line 2261 inside method `_updateKBArticle`:

```java
     _startWorkflowInstance(userId, kbArticle, serviceContext);
```

Reply:
yeah, that part is not needed neither for the common part.
I will extract from there.